sympy/utilities/lambdify.py

```diff
@@ -1119,7 +1119,21 @@ def doprint(self, funcname, args, expr, *, cses=()):
         if not iterable(args):
             args = [args]
 
-        argstrs, expr = self._preprocess(args, expr)
+        if cses:
+            subvars, subexprs = zip(*cses)
+            try:
+                exprs = expr + list(subexprs)
+            except TypeError:
+                try:
+                    exprs = expr + tuple(subexprs)
+                except TypeError:
+                    expr = [expr]
+                    exprs = expr + list(subexprs)
```

Comment:
I really can't understand what this is supposed to be doing. Here `expr` comes from the user and is the expression that they wanted to `lambdify`. This code seems to presume that `expr` is either a list or a tuple or otherwise it can be safely turned into a list.

Reply:
I wrote:

> This isn't an elegant fix, but seems to work.

above to warn about this. It was merged nonetheless.

My suggestion would be to enable `cse=True` for copies of all of the lambdify tests that cover the variety of input/output combinations to lambdify. More bugs may also be revealed (from the original addition of the cse flag and this PR).